pkg/test/api_server.go

```diff
@@ -42,9 +43,21 @@ func NewAPIServer(t *testing.T) *APIServer {
 	t.Helper()
 
 	var (
-		veleroClient    = fake.NewSimpleClientset()
-		kubeClient      = kubefake.NewSimpleClientset()
-		dynamicClient   = dynamicfake.NewSimpleDynamicClient(runtime.NewScheme())
+		veleroClient  = fake.NewSimpleClientset()
+		kubeClient    = kubefake.NewSimpleClientset()
+		dynamicClient = dynamicfake.NewSimpleDynamicClientWithCustomListKinds(runtime.NewScheme(),
+			map[schema.GroupVersionResource]string{
```

Comment:
Is this update due to the library upgrade as well? I don't think this is related, but we have some other places where we add to the scheme, do you know if those are affected by the upgrade as well? https://github.com/vmware-tanzu/velero/blob/main/pkg/client/factory.go#L156-L160

Reply:
I had to make this change because I saw the UT failure after bumping up `client-go`, similar to https://github.com/kubernetes/client-go/issues/914

Reply:
Thanks :+1: Looking at this again, I don't think the second part of my comment makes sense. I see that we just need to pass in all the resources we want to list with this fake client. I thought it was maybe related to setting up the scheme like we do elsewhere.